### PR TITLE
New R Cheat Sheet

### PR DESCRIPTION
**Pull request by laurenancona**
https://duck.co/ia/view/r_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@laurenancona Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/r.json [hunk 1]
```diff
@@ -0,0 +1,212 @@
+{
+  "id": "r_cheat_sheet",
+  "name": "R Cheatsheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/r.json]: Usually we don't put `Cheat Sheet` in the title. Something like "R Lang" or "R Methods" would be good though :+1:

Comment (laurenancona) [inline share/goodie/cheat_sheets/json/r.json]: @moollaza Title == `name`? 

Thanks!

File: share/goodie/cheat_sheets/json/r.json [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  "id": "r_cheat_sheet",
+  "name": "R Language",
+  "description": "Basic functions available in R base package",
+  "metadata": {
+    "sourceName": "R Reference Card - CRAN",
+    "sourceUrl": "https://cran.r-project.org/doc/contrib/Short-refcard.pdf"
+  },
+  "aliases": ["r language"],
+  "template_type": "terminal",
+  "section_order": ["Basic Syntax", "Input and output", "Data creation", "Data selection and manipulation", "Math", "Strings", "Dates and Times", "Plotting", "Getting help"],
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/r.json]: Lets format this a bit - have it dangle.

File: share/goodie/cheat_sheets/json/r.json [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+      "key": "strsplit(x,split)",
+      "val": "split x by the substring \"split\""
+    }, {
+      "key": "grep(pattern,x)",
+      "val": "searches for matches to pattern within x"
+    }, {
+      "key": "gsub(pattern,replacement,x)",
+      "val": "replaces all match occurances determined by regular expression"
+    }],
+    "Dates and Times": [{
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/r.json]: Lets use `Date and Time`.

File: share/goodie/cheat_sheets/json/r.json [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+      "val": "convert to uppercase"
+    }, {
+      "key": "strsplit(x,split)",
+      "val": "split x by the substring \"split\""
+    }, {
+      "key": "grep(pattern,x)",
+      "val": "searches for matches to pattern within x"
+    }, {
+      "key": "gsub(pattern,replacement,x)",
+      "val": "replaces all match occurances determined by regular expression"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/r.json]: `occurrences`

File: share/goodie/cheat_sheets/json/r.json [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+      "key": "toupper(x)",
+      "val": "convert to uppercase"
+    }, {
+      "key": "strsplit(x,split)",
+      "val": "split x by the substring \"split\""
+    }, {
+      "key": "grep(pattern,x)",
+      "val": "searches for matches to pattern within x"
+    }, {
+      "key": "gsub(pattern,replacement,x)",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/r.json]: In this (and those above) lets put a space after the commas, so `pattern, replacement` etc.

File: share/goodie/cheat_sheets/json/r.json [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+      "key": "/",
+      "val": "division"
+    }, {
+      "key": "%/%",
+      "val": "integer division"
+    }, {
+      "key": "%%",
+      "val": "remainder"
+    }],
+    "Input and output": [{
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/r.json]: Lets use `Input and Output` (remember to update `section_order`).

File: share/goodie/cheat_sheets/json/r.json [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+      "key": "read.table(\"filename\", sep=\",\")",
+      "val": "reads information in from file with a variable delimiter"
+    }, {
+      "key": "read.csv(\"filename\",header=TRUE)",
+      "val": "read csv file into dataframe"
+    }, {
+      "key": "write.csv(\"filename\")",
+      "val": "writes a dataframe or matrix (or tries to convert input to one) and saves to csv"
+    }],
+    "Data creation": [{
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/r.json]: Lets use `Data Creation` (and in `section_order`).

File: share/goodie/cheat_sheets/json/r.json [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+    }],
+    "Basic Syntax": [{
+      "key": "#",
+      "val": "comments"
+    }, {
+      "key": "<- or =",
+      "val": "assignment operator"
+    }, {
+      "key": "<<-",
+      "val": "global assignment operator"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/r.json]: In this and that above, I think it reads better without `operator`.

File: share/goodie/cheat_sheets/json/r.json [hunk 9]
```diff
@@ -0,0 +1,220 @@
+{
+  "id": "r_cheat_sheet",
+  "name": "R Language",
+  "description": "Basic functions available in R base package",
+  "metadata": {
+    "sourceName": "R Reference Card - CRAN",
+    "sourceUrl": "https://cran.r-project.org/doc/contrib/Short-refcard.pdf"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/r.json]: Hmm... I'm getting an invalid certificate from this site, do you have the same issue?

File: share/goodie/cheat_sheets/json/r.json [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+  },
+  "aliases": ["r language"],
+  "template_type": "terminal",
+  "section_order": ["Basic Syntax",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/r.json]: Lets pull this down to be aligned with the others.

File: share/goodie/cheat_sheets/json/r.json [hunk 11]
```diff
@@ -0,0 +1,220 @@
...
+      "val": "matrix multiplication"
+    }, {
+      "key": "/",
+      "val": "division"
+    }, {
+      "key": "%/%",
+      "val": "integer division"
+    }, {
+      "key": "%%",
+      "val": "remainder"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/r.json]: Do you mean `modulo`?

File: share/goodie/cheat_sheets/json/r.json [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+    }, {
+      "key": "help.start()",
+      "val": "start the HTML version of help"
+    }, {
+      "key": "str(a)",
+      "val": "display the internal structure of an R object"
+    }],
+    "Basic Syntax": [{
+      "key": "#",
+      "val": "comments"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/r.json]: Just `comment`.

**Comment by moollaza** [comment]
@laurenancona Thanks! This LGTM :+1:

**Comment by laurenancona** [comment]
Argh. Looks like some square brackets are being dropped in UI:
screen shot 2016-03-11 at 10 44 06 am
screen shot 2016-03-11 at 10 42 33 am

Tried various escapes but all produced errors in test env. 
@moollaza any thoughts?

**Comment by GuiltyDolphin** [comment]
@laurenancona Doh! We need to [double-escape the brackets](http://docs.duckduckhack.com/frontend-reference/cheat-sheet-reference.html#code-blocks-in-the-key-property)!

**Comment by laurenancona** [comment]
Weird that it only affects when there is a space present. Updating.

**Pull request merged**